src/main/scala/beam/agentsim/agents/PersonAgent.scala

```diff
@@ -587,15 +614,10 @@ class PersonAgent(
         val nextCoord = nextActivity(data).get.getCoord
         goto(ChoosingMode) using ChoosesModeData(
           personData = data.copy(
-            // If the mode of the next leg is defined and is CAV, use it, otherwise,
-            // If we don't have a current tour mode (i.e. are not on a tour aka at home),
-            // use the mode of the next leg as the new tour mode.
-            currentTourMode = modeOfNextLeg match {
-              case Some(CAV) =>
-                Some(CAV)
-              case _ =>
-                data.currentTourMode.orElse(modeOfNextLeg)
-            },
+            // We do not stick to current tour mode
```

Comment:
I'm wondering if here we can make use of the `currentTour` method defined above to take advantage of the tour structure already in BeamPlan.scala. It might take some archaeology to figure out how exactly everything was intended to function, but it looks like the `strategies` map in a `BeamPlan` allows for modes to be stored both for individual trips/legs and for tours.

So, I'm wondering if something like

`val tourModeOption = _experiencedBeamPlan.getStrategy(currentTour(data), classOf[ModeChoiceStrategy])`

would get the current tour mode, and we could do something similar with `putStrategy` when the initial trip mode is chosen and we want to set the tour mode.

Reply:
Replayed in the comment bellow.